jetbrains-ultimate/tst-201/software/aws/toolkits/jetbrains/services/redshift/RedshiftExplorerNodeTest.kt

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.redshift
+
+import com.intellij.testFramework.ProjectRule
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import software.amazon.awssdk.services.redshift.model.Cluster
+import software.amazon.awssdk.utils.CompletableFutureUtils
+import software.aws.toolkits.core.utils.RuleUtils
+import software.aws.toolkits.jetbrains.core.MockResourceCache
+import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerEmptyNode
+import software.aws.toolkits.jetbrains.core.explorer.nodes.AwsExplorerErrorNode
+import software.aws.toolkits.jetbrains.core.explorer.nodes.RedshiftExplorerRootNode
+
+class RedshiftExplorerNodeTest {
+    @JvmField
+    @Rule
+    val projectRule = ProjectRule()
+
+    private fun resourceCache() = MockResourceCache.getInstance(projectRule.project)
```

Comment:
Would you mind creating a new `ResourceCacheRule` test rule similar to [`MockClientManager`](https://github.com/aws/aws-toolkit-jetbrains/blob/d75ddd35d87e108f03ce19a1f6ea5c602f4e5aed/jetbrains-core/tst/software/aws/toolkits/jetbrains/core/MockClientManager.kt#L59) - I'm hoping to gradually deprecate direct use of the "Mock" instances - and replace them with rules so that we can better control their lifecycle.

Reply:
Makes a lot of sense, we have this duplicated around way too many places. Added